fabric-containers-v0/src/main/java/net/fabricmc/fabric/impl/client/container/ScreenProviderRegistryImpl.java

```diff
@@ -55,7 +50,7 @@ public void registerFactory(Identifier identifier, ContainerFactory<HandledScree
 	@Override
 	public <C extends ScreenHandler> void registerFactory(Identifier identifier, ContainerScreenFactory<C> containerScreenFactory) {
 		registerFactory(identifier, (syncId, identifier1, player, buf) -> {
-			C container = ContainerProviderImpl.INSTANCE.createContainer(syncId, identifier1, player, buf);
+			C container = ((ContainerProviderImpl) ContainerProviderImpl.INSTANCE).createContainer(syncId, identifier1, player, buf);
```

Comment:
Shouldn't this be `((ContainerProviderImpl) ContainerProviderRegistry.INSTANCE)`, I think this either compiles to that or it might either cause the same issues

Reply:
It's basically the same, only reason I didnt do it was becuase I am lazy, but I dont think it makes any diffrence.

I have tested this PR in AOF and it seems to work fine.

Reply:
Yeah there are many other places in this PR using this pattern. I think it just compiles to the same since the impls don't have their own fields, but that's a bit unclear.

Reply:
Quoting https://docs.oracle.com/javase/specs/jls/se8/html/jls-12.html#jls-12.4.1:

> A reference to a static field (§8.3.1.1) causes initialization of only the class or interface that actually declares it, even though it might be referred to through the name of a subclass, a subinterface, or a class that implements an interface.

So it's fine to write it like this, it's just a little confusing to look at.

Reply:
Yeah, just a bit confusing, it is fine I guess.

Reply:
Cleaned it up, found another case of it as well. 👍